src/storage/v2/disk/storage.cpp

```diff
@@ -69,6 +69,31 @@
 #include "utils/string.hpp"
 #include "utils/typeinfo.hpp"
 
+namespace {
+
+memgraph::storage::EdgeDirection ApproximateSmallerDegree(memgraph::storage::Vertex *from_vertex,
+                                                          memgraph::storage::Vertex *to_vertex) {
```

Comment:
> We are not approximating anything

Its `Approximate` because it uses the current edge count, it is completely timestamp and hence also delta agnostic. Its not perfect but a good heuristic approach to select the smaller side.

> If the function returns EdgeDirection::OUT that means that the from_vertex has less outgoing edges than the to_vertex has incoming ones right?

Yes

> Then, on the caller side we .... Why not just return the result of that VertexAccessor member (either InEdges or OutEdges) ...

Its only after refactoring the call site got that simple. You are right we can move that bit here. Just need to find a good name for the function.